scanpy/preprocessing/combat.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+from scipy.sparse import issparse
+import pandas as pd
+import sys
+from numpy import linalg as la
+import patsy
+
+def design_mat(mod, batch_levels):
+    # require levels to make sure they are in the same order as we use in the
+    # rest of the script.
+    design = patsy.dmatrix("~ 0 + C(batch, levels=%s)" % str(batch_levels),
```

Comment:
We'd usually use `.format` instead of `%`, but that's just a suggestion...

Reply:
thanks, did that!